pandas/tools/pivot.py

```diff
@@ -436,18 +446,76 @@ def crosstab(index, columns, values=None, rownames=None, colnames=None,
     data.update(zip(rownames, index))
     data.update(zip(colnames, columns))
 
+    if values is None and aggfunc is not None:
+        raise ValueError("aggfunc cannot be used without values.")
+
+    if aggfunc is not None and normalize is not False:
+        raise ValueError("Cannot use normalize with aggfunc")
+
     if values is None:
         df = DataFrame(data)
         df['__dummy__'] = 0
         table = df.pivot_table('__dummy__', index=rownames, columns=colnames,
                                aggfunc=len, margins=margins, dropna=dropna)
-        return table.fillna(0).astype(np.int64)
+        table = table.fillna(0).astype(np.int64)
+
     else:
         data['__dummy__'] = values
         df = DataFrame(data)
         table = df.pivot_table('__dummy__', index=rownames, columns=colnames,
                                aggfunc=aggfunc, margins=margins, dropna=dropna)
-        return table
+
+    # Post-process
+    table = _normalize(table, normalize=normalize, margins=margins)
+
+    return table
+
+
+def _normalize(table, normalize, margins):
+
+    if not isinstance(normalize, bool) and not isinstance(normalize, str):
```

Comment:
476-481 were to deal with the 1s and 0s. I tried just putting them into the `normalizers` dict, but dicts treat `True` and `1` (and `False` and `0`) as the same when used as keys, so I had to deal with converting 1s and 0s before the dict. Otherwise if I try and do:

     normalizer[0] = normalizer['index']

I end up overwriting the entry for `normalizer[False]` and everything goes to haywire.